src/GafferSceneUI/SceneView.cpp

```diff
@@ -1810,7 +1810,10 @@ void SceneView::frame( const PathMatcher &filter, const Imath::V3f &direction )
 
 void SceneView::expandSelection( size_t depth )
 {
-	Context::Scope scope( getContext() );
+	// Create a copy of the current context to evaluate the scene in, so that we aren't using the
+	// current context while we're modifying it.
+	Context referenceContext( *getContext(), Context::Copied );
```

Comment:
This should be `ContextPtr .. = new Context()`. Constructing a RefCounted object on the stack is asking for trouble, because it'll get double destructed if something else decides to add a reference (for instance, to pass it to Python).

I'm a little bit worried about the performance consequences here though, because it'll cause a copy-on-write for the modifications we subsequently make to the expanded paths. Let's chat about that in our catch up today.